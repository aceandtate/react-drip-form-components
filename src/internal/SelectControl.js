import styled from 'styled-components';

import TextControl from './TextControl';
import { get } from '../theme';

const Control = TextControl.withComponent('select');

const SelectControl = styled(Control)`
  height: ${props => props.multiple
    ? 'auto'
    : get(props, 'controlHeight')};
`;

export default SelectControl;
