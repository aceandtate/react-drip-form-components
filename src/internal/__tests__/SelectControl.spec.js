import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import SelectControl from '../SelectControl';
import { mountWithTheme } from '../../__tests__/utils';


describe('Internal#<SelectControl />', () => {
  test('Should be styled component', () => {
    const wrapper = shallow(<SelectControl />);
    expect(wrapper).toMatchSnapshot();
  });


  test('Should be auto height if specified multiple prop', () => {
    const wrapper = mountWithTheme(<SelectControl />, { controlHeight: '5em' });
    expect(wrapper).toHaveStyleRule('height', '5em');
  });
});
