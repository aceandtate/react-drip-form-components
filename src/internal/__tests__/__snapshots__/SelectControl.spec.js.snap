// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Internal#<SelectControl /> Should be styled component 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  width: 100%;
  height: 2.7em;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  margin: 0;
  padding: 0.3em 0.7em;
  background: rgb(255,255,255);
  border-style: solid;
  border-width: 1px;
  border-color: rgb(219,219,219);
  border-radius: 3px;
  box-shadow: 0 0 0 transparent;
  color: rgb(54,54,54);
  vertical-align: top;
  font-size: 1rem;
  font-family: inherit;
  line-height: 1.4;
  -webkit-transition-property: color,box-shadow,border;
  transition-property: color,box-shadow,border;
  -webkit-transition-duration: 110ms;
  transition-duration: 110ms;
  -webkit-transition-timing-function: cubic-bezier(.25,.46,.45,.94);
  transition-timing-function: cubic-bezier(.25,.46,.45,.94);
  height: 2.7em;
}

.c0:not(:disabled):hover {
  background: rgb(255,255,255);
  border-color: rgb(204,204,204);
}

.c0:not(:disabled):focus {
  outline: none;
  background: rgb(255,255,255);
  border-color: rgb(0,188,212);
  box-shadow: 0 2px 2px rgba(0,0,0,0.04) inset;
}

.c0:disabled {
  color: rgb(204,204,204);
  background: rgb(245,245,245);
  border-color: rgb(245,245,245);
  cursor: not-allowed;
}

<select
  className="c0"
/>
`;
